### PR TITLE
Update knative function controller

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- fix mac build, change resource names 

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/4803
kyma-project/console#804

### CODE REVIEW

File: components/knative-function-controller/README.md [hunk 1]
```diff
@@ -18,13 +18,39 @@ This Knative-based serverless implementation defines and handles the Function Cu
 ### Run locally
 Follow these steps to run the Knative Function controller locally:
 
-1. Modify the `config/config.yaml` file to include your base64-encoded `gcr.io` or `docker.io` credentials. 
-2. Update the Docker registry value to your `docker.io` username.
+1. Create a ServiceAccount to enable knativeBuild docker builds
```
Comment (bszwarc) [inline components/knative-function-controller/README.md]:
```suggestion
1. Create a ServiceAccount to enable knativeBuild Docker builds.
```

File: components/knative-function-controller/README.md [hunk 2]
```diff
@@ -18,13 +18,39 @@ This Knative-based serverless implementation defines and handles the Function Cu
...
 
-3. Apply the configuration:
+Currently you cannot run builds in every existing namespace of your cluster. ServiceAccounts with linked docker
```
Comment (bszwarc) [inline components/knative-function-controller/README.md]:
```suggestion
Currently, you cannot run builds for every Namespace existing in your cluster. 
```

File: components/knative-function-controller/README.md [hunk 3]
```diff
@@ -18,13 +18,39 @@ This Knative-based serverless implementation defines and handles the Function Cu
 ### Run locally
 Follow these steps to run the Knative Function controller locally:
 
-1. Modify the `config/config.yaml` file to include your base64-encoded `gcr.io` or `docker.io` credentials. 
-2. Update the Docker registry value to your `docker.io` username.
+1. Create a ServiceAccount to enable knativeBuild docker builds
 
-3. Apply the configuration:
+Currently you cannot run builds in every existing namespace of your cluster. ServiceAccounts with linked docker 
+repository credentials have to be created for each namespace that will be used for the knative-function controller.
```
Comment (bszwarc) [inline components/knative-function-controller/README.md]:
```suggestion
Create ServiceAccounts with linked Docker repository credentials for each Namespace that will be used for the knative-function controller.
```

File: components/knative-function-controller/README.md (hunk 2 above)
Comment (k15r) [reply components/knative-function-controller/README.md]: the build cannot run if it is created IN a certain namespace, as it requires a serviceaccount in that namespace

Comment (bszwarc) [reply components/knative-function-controller/README.md]: Maybe we can combine these two and have something like this then:

`To run a build in a Namespace, create ServiceAccounts with linked Docker repository credentials for each Namespace that will be used for the knative-function controller.`

**Comment by rakesh-garimella** [comment]
closing this PR for: https://github.com/kyma-project/kyma/pull/5333

**Pull request closed without merging**